isb_lib/geome_adapter.py

```diff
@@ -96,39 +170,68 @@ def listProjects(self):
-                )
-                break
-            # L.debug("recordsInProject data: %s", response.text[:256])
-            data = response.json()
-            for record in data.get("content", {}).get(record_type, []):
-                L.debug("recordsInProject Record id: %s", record.get("bcid", None))
-                # print(json.dumps(record, indent=2))
-                # raise NotImplementedError
-                yield record
-            if len(data.get("content", {}).get(record_type, [])) < _page_size:
-                more_work = False
-            params["page"] = params["page"] + 1
+            return
+        # L.debug("recordsInProject data: %s", response.text[:256])
+        expeditions_json = response.json()
+        if len(expeditions_json) == 0:
+            L.error("Project doesn't have any expeditions project:%s", project_id)
+        for expedition_dict in expeditions_json:
+            # Skip developer messages
+            if type(expedition_dict) is dict:
+                expedition_modified_date = expedition_dict["modified"]
+                expedition_modified_datetime = dateparser.parse(expedition_modified_date)
+                have_data_for_project = str(project_id) in self.known_existing_projects
+                # For each expedition, check to see if the mod date is greater than the last time we fetched.
+                # If it is, fetch the bcids of the samples in the expedition.
+                if expedition_modified_datetime.timestamp() > self._date_start.timestamp(): # or not have_data_for_project:
+                    more_work = True
+                    params = {
+                        "limit": _page_size,
+                        "page": 0,
+                        "q": f"_expeditions_:{expedition_dict['expeditionCode']}",
+                    }
+                    while more_work:
+                        expedition_records_url = f"{GEOME_API}records/Sample/json"
+                        response = requests.get(
+                            expedition_records_url, params=params, headers=headers, timeout=HTTP_TIMEOUT
+                        )
+                        if response.status_code != 200:
+                            L.error(
+                                "Unable to load project expeditions project:%s; expedition:%s, status: %s; reason: %s",
+                                project_id,
+                                expedition_dict["expeditionCode"],
+                                response.status_code,
+                                response.reason,
+                            )
+                            break
+                        # L.debug("recordsInProject data: %s", response.text[:256])
+                        expeditions_json = response.json()
+
+                        for record in expeditions_json.get("content", {}).get(record_type, []):
+                            L.debug("recordsInProject Record id: %s", record.get("bcid", None))
```

Comment:
This piece is somewhat interesting in that we're now recording the mod date of the records as the mod date of the expedition.  Otherwise, we could end up refetching things.